src/main/java/io/inscopemetrics/kairosdb/aggregators/HistogramFilterAggregator.java

```diff
@@ -200,47 +177,66 @@ private void moveCurrentDataPoint() {
             double sum = 0;
             long count = 0;
-            long originalCount = 0;
 
-            if (dp instanceof HistogramDataPoint) {
-                final HistogramDataPoint hist = (HistogramDataPoint) dp;
-                originalCount = hist.getOriginalCount();
+            final HistogramDataPoint hist = (HistogramDataPoint) dp;
+            final HistogramKeyUtility histogramKeyUtility = HistogramKeyUtility.getInstance(hist.getPrecision());
+            final long originalCount = hist.getOriginalCount();
 
-                if (histNotChangedByThreshold(hist)) {
-                    return hist;
-                } else {
-                    for (final Map.Entry<Double, Integer> entry : hist.getMap().entrySet()) {
-                        if (!shouldDiscard(entry.getKey())) {
-                            filtered.put(entry.getKey(), entry.getValue());
-                            min = Math.min(min, Math.min(entry.getKey(), binInclusiveBound(entry.getKey())));
-                            max = Math.max(max, Math.max(entry.getKey(), binInclusiveBound(entry.getKey())));
-                            sum += entry.getKey() * entry.getValue();
-                            count += entry.getValue();
-                        }
-                    }
-                    if (minNotChangedByThreshold(hist)) {
-                        min = hist.getMin();
-                    }
-                    if (maxNotChangedByThreshold(hist)) {
-                        max = hist.getMax();
-                    }
+            if (histNotChangedByThreshold(hist)) {
+                return Optional.of(hist);
+            }
+
+            for (final Map.Entry<Double, Long> entry : hist.getMap().entrySet()) {
+                if (!shouldDiscard(entry.getKey(), histogramKeyUtility)) {
+                    filtered.put(entry.getKey(), entry.getValue());
+                    min = Math.min(
+                            min,
+                            Math.min(
+                                    entry.getKey(),
+                                    histogramKeyUtility.binInclusiveBound(entry.getKey())));
+                    max = Math.max(
+                            max,
+                            Math.max(
+                                    entry.getKey(),
+                                    histogramKeyUtility.binInclusiveBound(entry.getKey())));
+                    sum += entry.getKey() * entry.getValue();
+                    count += entry.getValue();
                 }
             }
-            return new HistogramDataPointImpl(
-                    timeStamp,
-                    filtered,
-                    min,
-                    max,
-                    sum / count,
-                    sum,
-                    originalCount);
+
+            if (filtered.isEmpty()) {
+                return Optional.empty();
+            }
+
+            // TODO(ville): If we know we're not going to change it then don't compute it above.
```

Comment:
resolve before merge?

Reply:
Unlikely, it's just a note about the code quality/performance that can be deferred. Let me know if you feel strongly otherwise.